mev_inspect/inspector.py

```diff
@@ -0,0 +1,82 @@
+import asyncio
+import logging
+import sys
+import traceback
+from asyncio import CancelledError
+
+from web3 import Web3
+from web3.eth import AsyncEth
+
+from mev_inspect.block import create_from_block_number
+from mev_inspect.classifiers.trace import TraceClassifier
+from mev_inspect.db import get_inspect_session, get_trace_session
+from mev_inspect.inspect_block import inspect_block
+from mev_inspect.provider import get_base_provider
+
+logging.basicConfig(stream=sys.stdout, level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+class MEVInspector:
+    def __init__(
+        self,
+        rpc: str,
+        cache: bool = False,
+        max_concurrency: int = 1,
```

Comment:
Actually the [semaphore](https://docs.python.org/3/library/asyncio-sync.html#semaphore) still applies:

"This class has been implicitly getting the current running loop"

So regardless of which tasks have been scheduled, since we only have a single running event loop. Since we're not using thread/process based parallelism, we're safe here and the max_concurrency should be respected

Reply:
Ohhh sick! Makes sense that it wraps the whole loop